### ISSUE TITLE
SpeasyVariable does not check `VALIDMIN` and `VALIDMAX`

### ISSUE THREAD

**Issue by Beforerr** [SciQLop/speasy#161]
### Description

CDF actually contains this meta to confine data valid range (not `nan`). It would be nice this is validated in the library level not by user.

### What I Did

I have a simple function to validate this

```
def get_data_and_time(v: SpeasyVariable):
    v = v.replace_fillval_by_nan()
    data = v.values
    time = v.time
    v_valid_mins = v.meta.get("VALIDMIN", [])
    v_valid_maxs = v.meta.get("VALIDMAX", [])

    all_cond_axis = tuple(range(1, data.ndim))
    for v_valid_min in v_valid_mins:
        cond = (data >= v_valid_min).all(axis=all_cond_axis)
        data, time = data[cond], time[cond]
    for v_valid_max in v_valid_maxs:
        cond = (data <= v_valid_max).all(axis=all_cond_axis)
        data, time = data[cond], time[cond]
    return data, time
    ```

**Comment by jeandet** [comment]
@Beforerr, definitely something we can integrate into Speasy. To be sure, you want to filter out any non valid data points from  a Speasy variable?
If so, I we can easily add a method `only_valid_values`  with the same interface than `replace_fillval_by_nan`. 
Speasy variables Numpy support should make this simpler too, I have to check if Speasy variables behave well with comparisons and Numpy indexing.

**Comment by Beforerr** [comment]
Yes, it would be a great idea to add a method like this. Also, If I remember right, this actually applies to the local CDF files. When geting the data from the CDAS (not sure about other service), it actually already does something like that.

**Comment by jeandet** [comment]
@Beforerr  I've prepared a PR (#168 ) with two new methods, `clamp_with_nan` that replaces values outside of Valid range with NaNs and `sanitized` that drops problematic values based on booleans. 
I also extended Numpy support to allow comparison and indexing, so you can simply write something like [this](https://github.com/jeandet/speasy/blob/valid_min_and_valid_max/speasy/products/variable.py#L694]

**Comment by Beforerr** [comment]
Hi @jeandet Thanks for the update! The method `sanitized` and extended Numpy support looks really useful! I appreciate the effort you put into the PR.
